cmd/mv-main.go

```diff
@@ -285,6 +285,15 @@ func mainMove(ctx *cli.Context) error {
 	// check 'copy' cli arguments.
 	checkCopySyntax(ctx, encKeyDB, true)
 
+	if ctx.NArg() == 2 {
+		args := ctx.Args()
+		srcURL := args.Get(0)
+		dstURL := args.Get(1)
+		if srcURL == dstURL {
+			fatalIf(probe.NewError(errors.New("")), fmt.Sprintf("Source and destination urls cannot be the same: %v.", srcURL))
```

Comment:
You would need to ignore like current behaviour than throwing an error.
```bash
[bala@localhost mc]$ ./mc mv one d1 d1
one:                       2 B / 2 B ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 1.69 KiB/s 0s
[bala@localhost mc]$ ls d1
one
```

Reply:
@balamurugana The use case here is to fix is the scenario where both the destination and source are objects and not a folder

```
mc mv play/test/issue play/test/issue
```
If both the source and destination are folders, we are already displaying error

```
 $ mc mv play/test/inner/ play/test/inner -r 
mc: <ERROR> Moving a folder into itself is not allowed. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
```

Reply:
With the same argument, why can't we error out for `./mc mv file dir dir`?

Reply:
probably same thing should have been done in `mv` and `cp` earlier.